accepted/00000-enable-ansible-control-node-in-registered-system.md

```diff
@@ -0,0 +1,136 @@
+
+At this point, Uyuni doesn't know how to reach those systems, we only know their "fqdns" according to the Ansible inventory, but the Ansible inventory in the control node should contain information about how to reach them via ssh. The idea here is the following:
+
+In order to bootstrap and trigger the registration of these systems, we have two alternatives:
+
+#### Use "salt-ssh" from Uyuni directly to the "fqdns" of the Ansible managed clients:
+
+* We previously need to fetch the corresponding SSH key that is used to connect to this particular host by Ansible.
+* Then we would reuse this key as SSH credentials to trigger the bootstrap on that host.
+* Downside: Uyuni needs to have direct access to the Ansible managed clients to bootstrap
+
+#### Use "salt-ssh" from Uyuni using the Ansible control node as SSH proxy:
+
+* We do not necessary need to fetch the corresponding SSH key, but we need to have the Uyuni SaltSSH deployed on the Ansible control node to access it
+* The SSH key to connect the Ansible managed client referenced in the Ansible inventory is used to build the ProxyCommand to use in "salt-ssh".
+
+NOTE: It needs to be clarified in the documentation that `ansible_ssh_private_key_file` needs to be defined in the Ansible inventory of your Ansible control node.
+
+* Once we have this, we can trigger the bootstrap of those systems by calling "salt-ssh" like this:
+
+```console
+# salt-ssh --ssh-option='ProxyCommand="/usr/bin/ssh -i /srv/susemanager/salt/salt_ssh/mgr_ssh_id -o StrictHostKeyChecking=no -o User=root -W %h:%p uyuni-ansible-control-node.tf.local"' --roster=ansible --roster-file=/var/cache/ansible/ansible-inventory.yaml -N webserver1 state.apply certs,bootstrap pillar='{"mgr_server": "uyuni-srv.tf.local", "minion_id": "uyuni-ansible-sles15sp1-2.tf.local"}'
+```
+In esence, we apply the same states that we do at the time of Boostrapping a new minion via the UI, passing the necessary information as pillar data.
+
+* Another example of usage (without using the ansible roster/targetting):
+
+```console
+# salt-ssh --ssh-option='ProxyCommand="/usr/bin/ssh -i PATH_TO_SALT_SSH_KEY_IN_THE_UYUNI_SERVER -o StrictHostKeyChecking=no -W %h:%p uyuni-stable-ansible-controller.tf.local /usr/bin/ssh -i PATH_TO_THE_SSH_KEY_IN_ANSIBLE_CONTROL_NODE uyuni-stable-ansible-opensuse152-2.tf.local" -o StrictHostKeyChecking=no' uyuni-stable-ansible-opensuse152-2.tf.local test.ping
+```
+
+
+### Triggering playbook executions in an Ansible control node
+
+After exploring the Ansible control node, inventory and playbooks, we can allow Uyuni to trigger the execution of one of those playbooks via a simple Salt state to execute in that control node:
```

Comment:
Should we also provide an easy way to trigger this from the UI? E. g. by adding Ansible playbooks directly to configuration channels (without the Salt state in the middle, we can autogenerate that state)?

Reply:
Maybe not in the first step. The problem is, that playbooks have the targets where it gets applied included. This cannot be changed without editing playbooks which is not planned yet. This topic along would need an own RFC from my point of view.

Reply:
> Maybe not in the first step

OK

> The problem is, that playbooks have the targets where it gets applied included. This cannot be changed without editing playbooks which is not planned yet. This topic along would need an own RFC from my point of view.

OK. Maybe an opportunity to move the editor to VS Codium or alike :-)